### DISCUSSION TITLE
analyse and plot pulses raises error message

### DISCUSSION THREAD

**Discussion by DieGraefin** [ozel/DIY_particle_detector#24] [General]
Hi Together,

I downloaded the script and installed all python packages. Now I wanted it to run and I get error messages:
TypeError: float() argument must be a string or a number, not 'list'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File ".../diode_detector/analyse_and_plot_pulses.py", line 688, in <module>
    hb = wf.hexbin(np.ravel(xpulses), np.asarray(ypulses).ravel(), gridsize=188, cmap='Greens', mincnt=1, vmax=100)
  File "/usr/lib/python3/dist-packages/matplotlib/__init__.py", line 1814, in inner
    return func(ax, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/matplotlib/axes/_axes.py", line 4077, in hexbin
    x = np.array(x, float)
ValueError: setting an array element with a sequence.

I added some prints and in the code when it fills the xpluses it somehow adds "list(...,...,...,)"

I think the code needs to be changed as it was written for a certain version of matplotlib and numpy.
Does anyone have suggestions?

Thanks

**Comment by ozel** [comment]
As explained in the header of the script, the idea is to execute it only in chunks, as needed. 
Spyder, PyCharm and few other editors recognise the `# <codecell>` as separators and then only execute each block of code selectively via SHIFT+RETURN (depending on the editor). The workflow is a bit like a jupyter notebook, but more flexible!

That lower part of the script which throws an error for you, using wf.hexbins(), is an extra which can be easily skipped (comment it out or move the `if 0` from further down just above it). This code is not necessary to get energy spectra and may actually take a very long time to compute with the large reference measurements. Energy spectra are plotted in the code cells right above that one.

I am realising this is a way of using the script is a bit too unusual for some, especially beginners. I will upload soon an update that simplifies the file loading via a command line parameter and skip the unnecessary parts of the code.
I will also add a requirements.txt file listing all the modules. Python Version 3.6/3.7 or newer and it's corresponding versions of Mathplotlib should work fine, older ones could be problematic.

**Reply by ozel** [reply]
I just confirmed, that hexbin code cell is not working at the moment with the rest of the script. In my local copy, I have not used it in a while.
